gdc_client/common/config.py

```diff
@@ -1,11 +1,28 @@
+import argparse
 import logging
+import sys
 from ConfigParser import ConfigParser, NoOptionError, NoSectionError
 
 from gdc_client.defaults import (
     processes, USER_DEFAULT_CONFIG_LOCATION, HTTP_CHUNK_SIZE
 )
 
-log = logging.getLogger('gdc-client-config')
+log = logging.getLogger('gdc-client')
+
+# This will display the default configs in a INI-type format, so that users
+# will be able to copy and modify as needed
+DISPLAY_TEMPLATE = '[{}]\n{}\n'
+
+
+class GDCClientArgumentParser(argparse.ArgumentParser):
+    """This is a workaround introduced here https://groups.google.com/forum/#!topic/argparse-users/LazV_tEQvQw
+    which enables to print the full help message in case something went wrong
+    with argument parsing
+    """
+    def error(self, message):
+        self.print_help(sys.stderr)
+        sys.stderr.write('\ngdc-client error: {}\n'.format(message))
```

Comment:
Why stderr instead of logging?

Reply:
I only ask because I think the dtt-ui pays attention to this sort of thing.

https://github.com/NCI-GDC/dtt-ui/search?q=stdout&unscoped_q=stdout

Reply:
Oh, I see. I will take a look. I was afraid that logger might not be setup at this point.

Reply:
Oh I didn't consider that. This is probably ok then.